### PR TITLE
Working on #2972 Template Conversion to Twig Format (managegroup.php)

### PR DESCRIPTION
**Pull request by PaulBender**
#2972 for managegroup.php

### CODE REVIEW

File: managegroup.php [hunk 1]
```diff
@@ -195,38 +197,42 @@
...
+
+    $query = $db->query("
+        SELECT j.*, u.uid, u.username, u.postnum, u.regdate
+        FROM " . TABLE_PREFIX . "joinrequests j
+        LEFT JOIN " . TABLE_PREFIX . "users u ON (u.uid=j.uid)
+        WHERE j.gid='{$gid}' AND j.uid != 0
+        ORDER BY u.username ASC
+    ");
+    while ($user = $db->fetch_array($query)) {
+        $user['reason'] = htmlspecialchars_uni($user['reason']);
```
Comment (euantorano) [inline managegroup.php]: The use of `htmlspecialchars_uni()` here (and perhaps below?) will cause double escaping.

File: managegroup.php [hunk 2]
```diff
@@ -257,185 +263,141 @@
...
+    $plugins->run_hooks("managegroup_start");
+
+    $lang->members_of = $lang->sprintf($lang->members_of, htmlspecialchars_uni($usergroup['title']));
+    $lang->add_member = $lang->sprintf($lang->add_member, htmlspecialchars_uni($usergroup['title']));
+    $lang->invite_member = $lang->sprintf($lang->invite_member, htmlspecialchars_uni($usergroup['title']));
+    $usergroup['pendingrequests'] = false;
+
+    if ($usergroup['type'] == 5) {
+        $usergroup['usergrouptype'] = $lang->group_public_invite;
+    }
```
Comment (euantorano) [inline managegroup.php]: Should be `} else if (...`.

File: managegroup.php [hunk 3]
```diff
@@ -257,185 +263,141 @@
...
+
+    // Display group leaders (if there is any)
+    $query = $db->query("
+        SELECT g.*, u.username, u.usergroup, u.displaygroup
+        FROM " . TABLE_PREFIX . "groupleaders g
+        LEFT JOIN " . TABLE_PREFIX . "users u ON (u.uid=g.uid)
+        WHERE g.gid = '{$gid}'
+    ");
+
+    $leaders_array = array();
```
Comment (euantorano) [inline managegroup.php]: For consistency with the below `$leaders = []`, can you use square brackets here too to initialise the array?

File: managegroup.php [hunk 4]
```diff
@@ -257,185 +263,141 @@
...
+        $leader_count = $db->num_rows($query);
+        while ($leader = $db->fetch_array($query)) {
+            $leader['username'] = htmlspecialchars_uni($leader['username']);
+            $leader_name = format_name($leader['username'], $leader['usergroup'], $leader['displaygroup']);
+            $leader['profilelink'] = build_profile_link($leader_name, $leader['uid']);
+
+            $leaders_array[] = $leader['uid'];
+
+            // Get commas...
+            if ($loop != $leader_count) {
```
Comment (euantorano) [inline managegroup.php]: This logic could be done in the view itself, using [the `loop.last` variable](https://twig.symfony.com/doc/2.x/tags/for.html#the-loop-variable).

**Comment by PaulBender** [comment]
All changes have been made

**Pull request merged**